auth/basic/middleware.go

```diff
@@ -56,26 +56,31 @@ func parseBasicAuth(auth string) (username, password []byte, ok bool) {
 	return c[:s], c[s+1:], true
 }
 
+// Returns a hash of a given slice.
+func toHashSlice(s []byte) []byte {
+	hash := sha256.Sum256(s)
+	return hash[:]
+}
+
 // AuthMiddleware returns a Basic Authentication middleware for a particular user and password.
 func AuthMiddleware(requiredUser, requiredPassword, realm string) endpoint.Middleware {
-	requiredUserBytes := sha256.Sum256([]byte(requiredUser))
-	requiredPassBytes := sha256.Sum256([]byte(requiredPassword))
+	requiredUserBytes := toHashSlice([]byte(requiredUser))
+	requiredPasswordBytes := toHashSlice([]byte(requiredPassword))
 
 	return func(next endpoint.Endpoint) endpoint.Endpoint {
 		return func(ctx context.Context, request interface{}) (interface{}, error) {
 			auth := ctx.Value(httptransport.ContextKeyRequestAuthorization).(string)
-			givenUser, givenPass, ok := parseBasicAuth(auth)
+			givenUser, givenPassword, ok := parseBasicAuth(auth)
 			if !ok {
 				return nil, AuthError{realm}
 			}
 
 			// Equalize lengths of supplied and required credentials by hashing them.
```

Comment:
- [ ] Since this comment points out _what_ is happening but not why, consider removing this comment.